internal/builder/builder.go

```diff
@@ -3,12 +3,15 @@ package builder
 import (
 	"context"
 
+	bkclient "github.com/moby/buildkit/client"
+
 	"github.com/dominodatalab/forge/internal/builder/config"
 	"github.com/dominodatalab/forge/internal/builder/embedded"
 )
 
+
 type OCIImageBuilder interface {
-	BuildAndPush(context.Context, *config.BuildOptions) ([]string, error)
+	BuildAndPush(context.Context, *config.BuildOptions, func(chan *bkclient.SolveStatus) error) ([]string, error)
```

Comment:
Unfortunately, this change will binds any/every implementation to buildkit. I like the idea of using an `io.Writer` type but what do you think about this [alternative approach](https://github.com/dominodatalab/forge/pull/27) to accomplish the same thing?